source/agora/common/BanManager.d

```diff
@@ -124,47 +96,73 @@ public class BanManager
 
     ***************************************************************************/
 
-    public this (Config config, Clock clock, string banfile_path,
-        Logger logger = Logger(ThisModule)) @safe nothrow pure
+    public this (Config config, Clock clock,
+        ManagedDatabase db, Logger logger = Logger(ThisModule)) @safe nothrow
     {
         this.config = config;
         this.log = logger;
         this.clock = clock;
-        this.banfile_path = banfile_path;
-    }
 
-    /***************************************************************************
+        this.db = db;
 
-        Load the ban data from disk
+        create_tables();
```

Comment:
`this.`

But I don't see why it's a separate method ?

Reply:
I went with doing all database access in separate `@trusted` functions, rather than using inline trusted.

Reply:
Fair enough. Just the style fixes then.